File: str_practice_space_compress.py
input_str = "abc efg   hij"
list1 = input_str.split(' ')
list2 = []
for str1 in list1:
    if str1:
        list2.append(''.join(reversed(str1)))
output_str = ' '.join(list2)
print(output_str)

File: str_practice_space_compress.rs
fn main() {
    let s = "abc efg   hij";
    let v = s.split(' ')
    .filter(|w| { *w != ""} )
    .map(|w| { w.chars().rev().collect::<String>()})
    // .inspect(|w|{ println!("{}", w); })
    .collect::<Vec<_>>();
    let rs = v.join( " ");
    println!("{}", rs);
}